src/KinesisVideoProducer.cpp

```diff
@@ -156,7 +156,8 @@ shared_ptr<KinesisVideoStream> KinesisVideoProducer::createStreamSync(unique_ptr
 
 void KinesisVideoProducer::freeStream(std::shared_ptr<KinesisVideoStream> kinesis_video_stream) {
     if (nullptr == kinesis_video_stream) {
-        LOG_AND_THROW("Kinesis Video stream can't be null");
+        LOG_ERROR("Kinesis Video stream can't be null");
+        return;
```

Comment:
Might be worth looking into how these APIs can communicate a failure to the caller. Similar to the next comment.

Reply:
I would prefer setting these to STATUS return type, for all of them to be in line with the producer and PIC SDKs. But will tackle that in another PR.